### ISSUE TITLE
Severe overfitting issue: High training recall (95%) but very low test recall (9%) on CSCampus3D dataset

### ISSUE THREAD

**Issue by riuXtan5** [csiro-robotics/HOTFormerLoc#4]
Hi, I'm training HOTFormerLoc on the CSCampus3D dataset and experiencing a severe overfitting problem. The pretrained weights you provided work perfectly during testing, but when I train from scratch following the provided config, the model severely overfits. The model achieves very high recall on the training set but performs poorly on the test set.

<img width="1182" height="271" alt="Image" src="https://github.com/user-attachments/assets/3abd2128-cda0-4785-9b26-5037dc59ad87" />

**Comment by Grifent** [comment]
Thanks for raising this issue. Have you tried training for the full 300 epochs as specified in the config? In my experience, the training dynamics on CS-Campus3D are quite unique compared to other datasets. I've noticed test set performance remains quite low until at least ~100-150 epochs, where the network overcomes some local minima and starts to improve rapidly. I observed a similar behaviour when training MinkLoc3Dv2, so it seems to be a characteristic of the dataset, or potentially the way batches are sampled. I suspect offline hard negative mining would help alleviate this, but we stick with the sampling approach used in MinkLoc3Dv2 for simplicity.
